### ISSUE TITLE
Training / Latent Extraction Bug

### ISSUE THREAD

**Issue by pftq** [Tencent/HunyuanVideo-I2V#43]
The dataset.py used in the latent extraction has a bug that returns 4 values instead of 5.  This triggers another error in run.py that then throws an error for not enough values to unpack.

In dataset.py, find:
```
if len(batch_index) == 0:
    print(f"get video len=0, skip for {video_item['video_path']}")
    return None, None, False
```

Replace with:
```
# 20250322 pftq: Ensure get_item always returns 5 values to fix 'not enough values to unpack (expected 5, got 3)'
if len(batch_index) == 0:
    print(f"get video len=0, skip for {video_item['video_path']}")
    return None, video_item["videoid"], video_item["video_path"], video_item["prompt"], False
```

**Comment by TianQi-777** [comment]
Thank you for your suggestion, we have fixed this problem according to your pull request.

**Comment by pftq** [comment]
One other bug should be noted - the latent extraction does not handle files in different subfolders with the same name.
For example:
a/file.mp4
b/file.mp4

There will only be one file.npy in the latents

**Comment by pftq** [comment]
Getting multi-GPU issues with the training script as well.  Testing with 4xH100 with "export HOST_GPU_NUM=4".  It fails with file not found trying to open json_path/.ckpts in hyvideo/dataset/video_loader.py
```
        for json_file in json_files:
            with open(f"{data_jsons_path}/{json_file}", 'r', encoding='utf-8-sig') as file:
                data = json.load(file)
``

**Comment by Changlin-Lee** [comment]
> One other bug should be noted - the latent extraction does not handle files in different subfolders with the same name. For example: a/file.mp4 b/file.mp4
> 
> There will only be one file.npy in the latents

Thank you for your suggestion, we assume the training files video_id (file name in our case) are unique. We will remind it in Readme

**Comment by Changlin-Lee** [comment]
> Getting multi-GPU issues with the training script as well. Testing with 4xH100 with "export HOST_GPU_NUM=4". It fails with file not found trying to open json_path/.ckpts in hyvideo/dataset/video_loader.py
> 
> ```
>         for json_file in json_files:
>             with open(f"{data_jsons_path}/{json_file}", 'r', encoding='utf-8-sig') as file:
>                 data = json.load(file)
> ``
> ```

Hi, we do not figure out any code will save ckpt into data_jsons_path. The path is only used for saving data. Do you make it as your ckpt save path?

**Comment by pftq** [comment]
No I do not - this only happens with multi-gpu.  Same code/configuration/files on single GPU runs fine without this error, so something is appending .ckpts to the json_path scan when parallelized

**Comment by Changlin-Lee** [comment]
> No I do not - this only happens with multi-gpu. Same code/configuration/files on single GPU runs fine without this error, so something is appending .ckpts to the json_path scan when parallelized

Which command did you run to get this error?

**Comment by pftq** [comment]
```
export HOST_GPU_NUM=1
chmod +x ./hyvideo/hyvae_extract/start.sh
./hyvideo/hyvae_extract/start.sh
sh scripts/run_train_image2video_lora.sh
```

Inside the vae.yaml
```
vae_path: "./ckpts/hunyuan-video-i2v-720p/vae"
video_url_files: "./training/meta_file.list"
output_base_dir: "./latents"
sample_n_frames: 129
target_size: 
  - 480
  - 480
enable_multi_aspect_ratio: True
use_stride: True
```

Inside the .sh
```
...
params=" \
    --lr 1e-4 \
    --warmup-num-steps 200 \
    --global-seed 1024 \
    --tensorboard \
    --zero-stage 2 \
    --vae 884-16c-hy \
    --vae-precision fp16 \
    --vae-tiling \
    --denoise-type flow \
    --flow-reverse \
    --flow-shift 7.0 \
    --i2v-mode \
    --model HYVideo-T/2 \
    --video-micro-batch-size 1 \
    --gradient-checkpoint \
    --ckpt-every 300 \
    --embedded-cfg-scale 6.0 \
    --epochs 51 \
    --final-save \
    "
...
```

**Comment by pftq** [comment]
Disregard - multi-gpu works fine after switching to this line in the sh file:
```
# single node, multi gpu
deepspeed --include localhost:0,1,2,3,4,5,6,7 --master_addr "${CHIEF_IP}" \
# single node, single gpu
#deepspeed --include localhost:0 --master_addr "${CHIEF_IP}" \
```

That should be made more clear in the readme - it's a huge time saver for training.
